### PR TITLE
featured articles

### PR DESCRIPTION
**Pull request by jetstream0**

### CODE REVIEW

File: repldex/config/__init__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 	EDITOR_IDS = [int(item.split()[0]) for item in f.read().splitlines()]
 	with open(config_path + 'entry_approval.txt') as f:
 		approves = f.read()
-		APPROVAL_IDS = [int(item.split()[0]) for item in approves.splitlines()]
+		APPROVAL_IDS = [item.split()[0] for item in approves.splitlines()]
```
Comment (Codemonkey51) [inline repldex/config/__init__.py:28]: Bruah that int() is needed. Pls add back

File: repldex/config/__init__.py [hunk 2]
```diff
@@ -44,4 +44,7 @@
 	BLACKLIST_IDS = [int(item.split()[0]) for item in f.read().splitlines()]
 	BLACKLISTED_IDS = BLACKLIST_IDS
 
+with open(config_path + 'featured.txt') as f:
+	FEATURED = [int(item.split()[0]) for item in f.read().splitlines()]
```
Comment (Codemonkey51) [inline repldex/config/__init__.py:48]: Entry id's are strings (uuid's to be exact) not ints, int() would breakity break break it

**Review by Codemonkey51 — CHANGES_REQUESTED** [review]
hmm seems like a good idea, few things borken tho

File: repldex/config/__init__.py (hunk 1 above)
Comment (jetstream0) [reply repldex/config/__init__.py:28]: but it fails checks

Comment (jetstream0) [reply repldex/config/__init__.py:28]: ohh wait nevermind

Comment (Codemonkey51) [reply repldex/config/__init__.py]: Lmao

**Review by Codemonkey51 — COMMENTED** [review]
Looks good but I can't test locally rn, so someone else will have todo that

**Comment by jetstream0** [comment]
Oh yeah I have an instance running on repl.it can test later

**Review by Codemonkey51 — CHANGES_REQUESTED** [review]
Use mongo db pls

**Comment by jetstream0** [comment]
TODO
If anyone has any objections or proposed modifications to the checklist below, let me know as soon as possible otherwise I may not actually add your modification or consider your objection due to the amount of time it would waste.
- [ ] separate collection for featured articles (although if you guys really really want I can maybe consider adding it directly to the entry collection)
- [ ] command on discord to add featured articles
- [ ] command on discord to remove featured articles
- [ ] add current featured article to main page top (above search bar?)
- [ ] command to change current featured article on main page

**Comment by Codemonkey51** [comment]
@jetstream0 no separate collection needed just add a featured parameter, and search for entries where that is true

**Comment by jetstream0** [comment]
> 
> 
> @jetstream0 no separate collection needed just add a featured parameter, and search for entries where that is true

seems fair

**Comment by jetstream0** [comment]
should be done

**Comment by jetstream0** [comment]
Will further test, but should work niw

**Comment by jetstream0** [comment]
I've tested, this should now work...
I tested it here: https://repl.it/@rediar/Repldex-Featured-Articles-Test
So if there are differences in code between the pr and the repl pls let me know

**Review by minx28 — APPROVED** [review]
pog

**Review by Codemonkey51 — APPROVED** [review]

**Pull request merged**